### PR TITLE
chore(infra): add a seed job for building releases for ampd and fix naming for building main branch ecr image

### PR DESCRIPTION
**Pull request by talalashraf**
## Todos

- [ ] Unit tests
- [ ] Manual tests
- [ ] Documentation
- [ ] Connect epics/issues

## Other Notes

### CODE REVIEW

File: .github/workflows/build-ampd-main.yaml [hunk 1]
```diff
@@ -1,17 +1,14 @@
-name: Build and push latest docker image
+name: Amplifier - Build main branch
```
Comment (milapsheth) [inline .github/workflows/build-ampd-main.yaml:1]:
```suggestion
name: Amplifier - main branch auto-builder
```
perhaps to better distinguish auto build actions vs manually triggered actions?

File: .github/workflows/build-ampd-release.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        description: Github tag to release binaries for (reusing an existing tag will make the pipeline fail)
+        required: true
+        default: latest
+
+jobs:
+  release-binaries:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
```
Comment (milapsheth) [inline .github/workflows/build-ampd-release.yaml:16]: for release builds, we should fix the OS version
```suggestion
        os: [ubuntu-22.04, macos-12]
```

File: .github/workflows/build-ampd-main.yaml (hunk 1 above)
Comment (talalashraf) [reply .github/workflows/build-ampd-main.yaml:1]: its both auto build and can be manually triggered.

Comment (milapsheth) [reply .github/workflows/build-ampd-main.yaml:1]: Fair but the main purpose is auto build anyways

Comment (talalashraf) [reply .github/workflows/build-ampd-main.yaml:1]: yes but then ill get questions about how to trigger the auto builder manually.

Comment (cgorenflo) [reply .github/workflows/build-ampd-main.yaml:1]: I'm with Talal on this. If I see auto-builder I will ignore it in the search for the correct workflow

**Review by milapsheth — APPROVED** [review]

**Pull request merged**